### PR TITLE
Default to elapsed time where possible

### LINKED ISSUE #72: Profile non-cpu time?
**Feature Request**: It would be really helpful to support a mode where CPU and non-CPU time is measured.

**Details**: From `Rprof`: "How time is measured varies by platform: on a Unix-alike it is the CPU time of the R process, so for example excludes time when R is waiting for input or for processes run by system to return."

Therefore, consider:

```{r}
system.time(download.file(url = "https://raw.githubusercontent.com/rstudio/sparklyr/master/inst/extdata/install_spark.csv", destfile = tempfile()))
```

with output:

```
trying URL 'https://raw.githubusercontent.com/rstudio/sparklyr/master/inst/extdata/install_spark.csv'
Content type 'text/plain; charset=utf-8' length 2534 bytes
==================================================
downloaded 2534 bytes

   user  system elapsed 
  0.017   0.004   0.423 
```

However, profiling:

```{r}
profvis::profvis(download.file(url = "https://raw.githubusercontent.com/rstudio/sparklyr/master/inst/extdata/install_spark.csv", destfile = tempfile()))
```

yields, `20ms`:

screen shot 2016-11-08 at 11 30 22 am

while this is correct from `profvis` use of `Rprof`, R is often used to proxy to other processes that users would want to be able to profile as well. Including an option to include the whole execution time without excluding IO calls, would be really helpful. That said, this probably requires a change in `Rprof` and `R` itself. Never the less, opening this feature request to discuss this further.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #72

### CODE REVIEW

File: R/profvis.R [hunk 1]
```diff
@@ -163,7 +182,10 @@ profvis <- function(expr = NULL,
     }
     repeat {
       inject(Rprof(prof_output, !!!rprof_args))
-      with_profvis_handlers(expr)
+      cnd <- with_profvis_handlers(expr)
```
Comment (hadley) [inline R/profvis.R:185]: Not related to this PR but `with_profvis_handlers` captures interrupts so without this there's no way to terminate this loop.

File: R/profvis.R [hunk 2]
```diff
@@ -109,6 +117,16 @@ profvis <- function(expr = NULL,
     message("Intervals smaller than ~5ms will probably not result in accurate timings.")
   }
 
+  if (is.null(timing)) {
+    if (getRversion() >= "4.4.0" || Sys.info()$sysname == "Windows") {
```
Comment (lionel-) [inline R/profvis.R:121]: To fix the `$ operator is invalid for atomic vectors` error.

```suggestion
    if (getRversion() >= "4.4.0" || Sys.info()[["sysname"]] == "Windows") {
```

File: R/profvis.R [hunk 3]
```diff
@@ -151,7 +169,8 @@ profvis <- function(expr = NULL,
       interval = interval,
       line.profiling = TRUE,
       gc.profiling = TRUE,
-      memory.profiling = TRUE
+      memory.profiling = TRUE,
+      event = timing
```
Comment (lionel-) [inline R/profvis.R:173]: I think this needs to be passed conditionally on R being recent enough?

File: tests/testthat/test-profvis.R [hunk 4]
```diff
@@ -13,3 +13,13 @@ test_that("Irrelevant stack is trimmed from profiles (#123)", {
   out <- repro_profvis(f(), simplify = TRUE)
   expect_equal(profvis_modal_value(out$x$message$prof), "pause f")
 })
+
+test_that("defaults to elapsed timing", {
+  skip_on_cran()
```
[the remark this answers is not in the document]
Comment (hadley) [reply tests/testthat/test-profvis.R:18]: Added a helper to match `has_simplify()`

**Review by lionel- — APPROVED** [review]

**Pull request merged**